scanner/matcher.go

```diff
@@ -133,6 +133,20 @@ func (m MatchIssuerRegex) PrecertificateMatches(p *ct.Precertificate) bool {
 	return m.PrecertificateIssuerRegex.FindStringIndex(p.TBSCertificate.Issuer.CommonName) != nil
 }
 
+// MatchSCTTimestamp is a matcher which matches leaf entries with the specified Timestamp.
+type MatchSCTTimestamp struct {
+	Timestamp uint64
+}
+
+func (m MatchSCTTimestamp) Matches(leaf *ct.LeafEntry) bool {
+	entry, _ := ct.LogEntryFromLeaf(1, leaf)
+	if entry == nil {
+		// Can't validate if we can't parse
+		return false
+	}
+	return entry.Leaf.TimestampedEntry.Timestamp == m.Timestamp
```

Comment:
I kinda prefer keeping it separate for a couple of reasons:
 - one's really an error case, even though it can't return the error due to the interface
 - having it split means it's very quick to scan and grok vs the compound version.

I'll assume you're ok with that since you said it was optional :)

Reply:
Oke doke